js/foundation.offcanvas.js

```diff
@@ -65,9 +85,9 @@ class OffCanvas extends Plugin {
       overlay.setAttribute('class', 'js-off-canvas-overlay ' + overlayPosition);
       this.$overlay = $(overlay);
       if(overlayPosition === 'is-overlay-fixed') {
-        $('body').append(this.$overlay);
+        $(this.$overlay).insertAfter(this.$element);
```

Comment:
My concern about having the overlay close rather than after the body is that unless I'm mistaken any sorts of transforms applied up the tree will break the fixed positioning